pypdf/_page.py

```diff
@@ -1868,6 +1878,83 @@ def process_operation(operator: bytes, operands: List[Any]) -> None:
             visitor_text(text, memo_cm, memo_tm, cmap[3], font_size)
         return output
 
+    def _layout_mode_fonts(self) -> Dict[str, _layout_mode.Font]:
+        """
+        Get fonts formatted for "layout" mode text extraction
+
+        Returns:
+            Dict[str, Font]: dictionary of _layout_mode.Font instances keyed by font name
+        """
+        # Font retrieval logic adapted from pypdf.PageObject._extract_text()
+        objr: Any = self
+        while NameObject(PG.RESOURCES) not in objr:
+            objr = objr["/Parent"].get_object()
+        resources_dict: Any = objr[PG.RESOURCES]
+        fonts: Dict[str, _layout_mode.Font] = {}
+        if "/Font" in resources_dict and self.pdf is not None:
+            for font_name in resources_dict["/Font"]:
+                *cmap, font_dict_obj = build_char_map(font_name, 200.0, self)
+                font_dict = {
+                    k: self.pdf.get_object(v)
+                    if isinstance(v, IndirectObject)
+                    else [
+                        self.pdf.get_object(_v) if isinstance(_v, IndirectObject) else _v
+                        for _v in v
+                    ]
+                    if isinstance(v, ArrayObject)
+                    else v
+                    for k, v in font_dict_obj.items()
+                }
+                # mypy really sucks at unpacking
+                fonts[font_name] = _layout_mode.Font(*cmap, font_dict)  # type: ignore
+        return fonts
+
+    def _layout_mode_text(
+            self,
+            space_vertically: bool = True,
+            scale_weight: float = 1.25,
+            debug_path: Union[Path, None] = None,
```

Comment:
```suggestion
            debug_path: Optional[Path] = None,
```

Reply:
The two representations are equivalent, and given that the "standard" notation in more recent python versions (`Path | None`) more closely resembles the `Union` notation, I'd prefer to leave as is unless you've strong opinions on the matter.